src/workspace-data/data-table/shared/DataTable.js

```diff
@@ -612,16 +616,34 @@ const DataTable = (props) => {
                         style: { marginLeft: '1rem' },
                         text: entityAttributeText(dataInfo),
                       });
-                      const editLink =
-                        editable &&
-                        dataProvider.features.supportsEntityUpdating &&
-                        h(EditDataLink, {
-                          'aria-label': `Edit attribute ${attributeName} of ${entityType} ${entityName}`,
-                          'aria-haspopup': 'dialog',
-                          'aria-expanded': !!updatingEntity,
-                          onClick: () => setUpdatingEntity({ entityName, attributeName, attributeValue: dataInfo }),
-                        });
 
+                      let extraEditableCondition = false;
+                      if (dataProvider.providerName === wdsProviderName) {
+                        const attributes = entityMetadata[entityType].attributes;
+                        const attributeType = getAttributeType(attributeName, attributes, dataProvider);
+                        // this will make fields that are not supported to have the edit icon be disabled
+                        if (attributeType.type === undefined) {
+                          extraEditableCondition = true;
+                        }
+                      }
+                      const editLink = !extraEditableCondition
+                        ? editable &&
+                          dataProvider.features.supportsEntityUpdating &&
+                          h(EditDataLink, {
+                            'aria-label': `Edit attribute ${attributeName} of ${entityType} ${entityName}`,
+                            'aria-haspopup': 'dialog',
+                            'aria-expanded': !!updatingEntity,
+                            onClick: () => setUpdatingEntity({ entityName, attributeName, attributeValue: dataInfo }),
+                          })
+                        : editable &&
+                          dataProvider.features.supportsEntityUpdating &&
+                          h(EditDataLinkDisabled, {
```

Comment:
Does this need a dedicated component? Or can the same thing be accomplished by setting `disabled` and `tooltip` props on EditDataLink?

Reply:
I wasnt sure if it was okay to override tooltip like that, but yes, that most certainly is much cleaner!